api/swagger.yaml

```diff
@@ -0,0 +1,282 @@
+      schema:
+        type: string
+  schemas:
+    chat_list_response:
+      type: object
+      properties:
+        chats:
+          type: array
+          items:
+            $ref: '#/components/schemas/detailed_chat'
+    chat_creation:
+      type: object
+      properties:
+        user_ids:
+          type: array
+          items:
+            type: string
+          example: [ 123, 321 ]
+        type:
+          $ref: '#/components/schemas/chat_type'
+    chat_response:
+      $ref: '#/components/schemas/detailed_chat'
+    detailed_chat:
+      allOf:
+        - $ref: '#/components/schemas/base_chat'
+        - $ref: '#/components/schemas/message_list_response'
+        - type: object
+          properties:
+            last_message:
+              $ref: '#/components/schemas/message'
+    base_chat:
+      properties:
+        chat_id:
+          type: string
+          example: 543
+        user_ids:
+          type: array
+          description: Ids of the users participating in the chat
+          items:
+            type: string
+          example: [ 123, 321 ]
+        last_activity:
+          type: string
+          format: date-time
+          description: Datetime of the last activity from this chat
+        type:
+          $ref: '#/components/schemas/chat_type'
+    chat_type:
+      type: string
+      enum:
+        - group
+        - direct
+    new_message_list_response:
+      type: object
+      properties:
+        new_messages:
+          type: array
+          items:
+            allOf:
+              - type: object
+                properties:
+                  chat_id:
+                    type: string
+                    example: 123
+              - $ref: '#/components/schemas/message_list_response'
+    message_list_response:
+      type: object
+      properties:
+        messages:
+          type: array
+          items:
+            $ref: '#/components/schemas/message'
+    message:
+      allOf:
+        - $ref: '#/components/schemas/base_message'
+        - type: object
+          properties:
+            id:
+              type: string
+              description: id of the message
+              example: 1234
+            is_read:
+              type: boolean
+              example: true
+              description: |
+                Defines if the recipient has already read the message
+            is_new:
+              type: boolean
+              example: false
+              description: |
+                defines if this message was already loaded to the chat of the recipient. This happens either via long polling for new messages, by fetching all chats for the recipient or by fetching all messages for the specific chat between recipient and author.
+            recipients_notified:
+              type: boolean
+              description: |
+                If set to true the recipient was notified by email that this messgae was written for him.
```

Comment:
In kind of group chat: what does this mean?
That all recipients where notified (are even have read the message) as this also happens in  WhatsApp-Groups?
Would be interesting to see how we will implement this ;-)

Reply:
as mentioned above, it is only meant for the user who requests the messages, see above :)